sync/profile.go

```diff
@@ -1,45 +1,440 @@
+				Method:      http.MethodPost,
+				Field:       "allowReplace",
+				Description: "Allow replacing existing profiles.",
+			}, {
+				Method:      http.MethodPost,
+				Field:       "validate",
+				Description: "Validate only.",
+			}, {
+				Method:      http.MethodPost,
+				Field:       "reset",
+				Description: "Replace all existing settings.",
+			}, {
+				Method:      http.MethodPost,
+				Field:       "allowUnknown",
+				Description: "Allow importing of unknown values.",
+			}},
+		BelongsTo:  module,
+		StructFunc: handleImportProfile,
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func handleExportProfile(ar *api.Request) (data []byte, err error) {
+	var request *ProfileExportRequest
+
+	// Get parameters.
+	q := ar.URL.Query()
+	if len(q) > 0 {
+		request = &ProfileExportRequest{
+			ID: q.Get("id"),
+		}
+	} else {
+		request = &ProfileExportRequest{}
+		if err := json.Unmarshal(ar.InputData, request); err != nil {
+			return nil, fmt.Errorf("%w: failed to parse export request: %w", ErrExportFailed, err)
+		}
+	}
+
+	// Check parameters.
+	if request.ID == "" {
+		return nil, errors.New("missing parameters")
+	}
+
+	// Export.
+	export, err := ExportProfile(request.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return serializeExport(export, ar)
+}
+
+func handleImportProfile(ar *api.Request) (any, error) {
+	var request *ProfileImportRequest
+
+	// Get parameters.
+	q := ar.URL.Query()
+	if len(q) > 0 {
+		request = &ProfileImportRequest{
+			ImportRequest: ImportRequest{
+				ValidateOnly: q.Has("validate"),
+				RawExport:    string(ar.InputData),
+				RawMime:      ar.Header.Get("Content-Type"),
+			},
+			AllowUnknown: q.Has("allowUnknown"),
+			AllowReplace: q.Has("allowReplace"),
+		}
+	} else {
+		request = &ProfileImportRequest{}
+		if err := json.Unmarshal(ar.InputData, request); err != nil {
+			return nil, fmt.Errorf("%w: failed to parse import request: %w", ErrInvalidImportRequest, err)
+		}
+	}
+
+	// Check if we need to parse the export.
+	switch {
+	case request.Export != nil && request.RawExport != "":
+		return nil, fmt.Errorf("%w: both Export and RawExport are defined", ErrInvalidImportRequest)
+	case request.RawExport != "":
+		// Parse export.
+		export := &ProfileExport{}
+		if err := parseExport(&request.ImportRequest, export); err != nil {
+			return nil, err
+		}
+		request.Export = export
+	case request.Export != nil:
+		// Export is aleady parsed.
+	default:
+		return nil, ErrInvalidImportRequest
+	}
+
+	// Import.
+	return ImportProfile(request, profile.SourceLocal)
+}
+
+// ExportProfile exports a profile.
+func ExportProfile(scopedID string) (*ProfileExport, error) {
+	// Get Profile.
+	r, err := db.Get(profile.ProfilesDBPath + scopedID)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to find profile: %w", ErrTargetNotFound, err)
+	}
+	p, err := profile.EnsureProfile(r)
+	if err != nil {
+		return nil, fmt.Errorf("%w: failed to load profile: %w", ErrExportFailed, err)
+	}
```

Comment:
Shouldn't the profile get locked here?

Reply:
This is parsing a raw entry from the DB. Nobody has can have a lock on this.